packages/table/src/TableBody.tsx

```diff
@@ -1,17 +1,57 @@
-import React from 'react';
-import { useTableContext } from './TableContext';
+import React, { useMemo, useCallback } from 'react';
+import { getDataComparisonFunction, useTableContext } from './TableContext';
 import { TableProps } from './Table';
 
 type TableBodyProps<Shape> = Pick<TableProps<Shape>, 'children'>;
 
+/**
+ * Hook that avoids re-rendering children when the data used to
+ * render them hasn't changed, even if their ordering has changed.
+ */
+function useRenderedChildren<Props>(
+  propList: Array<Props>,
+  renderFunction: (props: Props) => React.ReactNode,
+  compareFn?: (a: Props, b: Props) => number,
+): Array<React.ReactNode> {
+  const pairs: Array<[Props, React.ReactNode]> = useMemo(
+    () =>
+      propList.map((props, index) => [
+        props,
+        <React.Fragment key={index}>{renderFunction(props)}</React.Fragment>,
+      ]),
+    [propList, renderFunction],
+  );
+
+  return useMemo(() => {
+    if (!compareFn) {
+      return pairs.map(([_, renderResult]) => renderResult);
```

Comment:
Can we think of a way to remove this additional loop?

Hosting two arrays of the datum and the renderResult?

Reply:
we would still need to build some form of mapping from props to renderResult in order to derive the sorted renderResults. I suppose I can rewrite this to use `Map` to make that more obvious.